static/js/src/support-chart.js

```diff
@@ -0,0 +1,550 @@
+    }
+  });
+}
+
+/**
+ *
+ * @param {*} svg
+ * @param {String} highlightVersion
+ * 
+ * Set version axis labels
+ */
+function highlightChartRow(svg, highlightVersion) {
+  svg.selectAll(".tick text").select(function () {
+    var text = this.textContent;
+    var isNotHighlightedVersion =
+      highlightVersion && !text.includes(highlightVersion);
+    var isYearLabel = text.includes("20") && !text.includes("Ubuntu ");
+
+    if (isNotHighlightedVersion) {
+      this.classList.add("chart__label--transparent");
+    }
+
+    if (isYearLabel) {
+      this.classList.remove("chart__label--transparent");
+    }
+  });
+}
+
+/**
+ *
+ * @param {*} svg
+ * 
+ * Set version axis labels
+ */
+function setVersionAxisLabels(svg, taskVersions) {
+  svg.selectAll(".version .tick text").select(function (tickLabel, index) {
+    this.textContent = taskVersions[index];
+  });
+}
+
+/**
+ *
+ * @param {*} svg
+ * @param {Int} height
+ *
+ * Adds vertical lines to the x axis
+ */
+function addXAxisVerticalLines(svg, height, margin) {
+  svg
+    .selectAll(".x.axis .tick line")
+    .attr("y1", height - margin.bottom - margin.top)
+    .attr("y2", 26);
+}
+
+/**
+ *
+ * @param {String} chartSelector
+ * @param {Object} taskStatus
+ *
+ * Builds key for supplied chart based on task status
+ */
+function buildChartKey(chartSelector, taskStatus) {
+  var taskStatusKeys = Object.keys(taskStatus);
+  var rectDimensions = 25;
+  var rowHeight = rectDimensions + 5;
+  var gapBetweenRectAndText = 10;
+  var verticalTextAlignmentOffset = 16;
+
+  var chartKey = d3
+    .select(chartSelector)
+    .append("svg")
+    .attr("class", "chart-key")
+    .attr("width", "550")
+    .attr("height", rowHeight * taskStatusKeys.length);
+
+  taskStatusKeys.forEach(function (key, i) {
+    var keyRow = chartKey
+      .append("g")
+      .attr("class", "chart-key__row")
+      .attr("transform", "translate(0, " + rowHeight * i + ")")
+      .attr("height", rectDimensions);
+
+    keyRow
+      .append("rect")
+      .attr("class", taskStatus[key])
+      .attr("width", rectDimensions)
+      .attr("height", rectDimensions)
+      .attr("y", 0);
+
+    keyRow
+      .append("text")
+      .text(formatKeyLabel(key))
+      .attr("class", "chart-key__label")
+      .attr("x", rectDimensions + gapBetweenRectAndText)
+      .attr("y", verticalTextAlignmentOffset);
+  });
+}
+
+/**
+ *
+ * @param {String} key
+ *
+ * Formats key into readable string
+ */
+function formatKeyLabel(key) {
+  var keyLowerCase = key.toLowerCase().replace(/_/g, " ");
+  var formattedKey =
+    keyLowerCase.charAt(0).toUpperCase() + keyLowerCase.substr(1);
+  formattedKey = formattedKey.replace(
+    "Lts",
+    "Ubuntu LTS release Standard Support"
+  );
+  formattedKey = formattedKey.replace(" openstack ", " OpenStack ");
+  formattedKey = formattedKey.replace("kub", "Kub");
+  formattedKey = formattedKey.replace(
+    "Interim release",
+    "Interim release standard security maintenance (9 months)"
```

Comment:
I believe the words need to be capitalized here as well, like the others

Reply:
What do you mean exactly?

Reply:
every first letter of each word should be capitalized, like the other keys below it, right?